### PR TITLE
Bring galaxy-test-api under CI also.

### PR DESCRIPTION
**Pull request by jmchilton**
Along the lines of #14011 and #13983.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: packages/test.sh [hunk 1]
```diff
@@ -53,8 +55,8 @@ for ((i=0; i<${#PACKAGE_DIRS[@]}; i++)); do
     pip install -r test-requirements.txt
 
     # Prevent execution of alembic/env.py at test collection stage (alembic.context not set)
-    # Also ignore ToolShed functional tests.
-    unit_extra='--doctest-modules --ignore galaxy/model/migrations/alembic/ --ignore=tool_shed/test/'
+    # Also ignore functional tests (galaxy_test and tool_shed/test/).
+    unit_extra='--doctest-modules --ignore=galaxy/model/migrations/alembic/ --ignore=tool_shed/test/  --ignore=galaxy_test/api --ignore=galaxy_test/selenium'
```
Comment (nsoranzo) [inline packages/test.sh]: There needs to be at least one unit or doc test per package, or pytest will fail below. This is why I added one doc test to both test_base and test_driver inside their `galaxy_test/` .

File: packages/test_api/requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 galaxy-test-base
+tuspy
```
Comment (nsoranzo) [inline packages/test_api/requirements.txt:8]:
```suggestion
pytest
python-dateutil
requests
tuspy
```

File: packages/test.sh (hunk 1 above)
Comment (nsoranzo) [reply packages/test.sh]: OK, I moved the doctests to unit tests and excluded the whole `galaxy_test/` again here. We just need to add some unit tests to `test_api` and `test_selenium` now.

Comment (jmchilton) [reply packages/test.sh]: It feels silly right?

Update: Should we catch the pytest exit code 5 (no tests run) and suppress it somehow?

Comment (nsoranzo) [reply packages/test.sh]: > It feels silly right?

If it doesn't have tests, it's probably broken 😄

> Update: Should we catch the pytest exit code 5 (no tests run) and suppress it somehow?

Yeah, that should be possible and a good workaround. Let me know if you'd like help with that.

File: packages/test.sh [hunk 3]
```diff
@@ -53,8 +55,8 @@ for ((i=0; i<${#PACKAGE_DIRS[@]}; i++)); do
     pip install -r test-requirements.txt
 
     # Prevent execution of alembic/env.py at test collection stage (alembic.context not set)
-    # Also ignore ToolShed functional tests.
-    unit_extra='--doctest-modules --ignore galaxy/model/migrations/alembic/ --ignore=tool_shed/test/'
+    # Also ignore functional tests (galaxy_test/ and tool_shed/test/).
+    unit_extra='--doctest-modules --ignore=galaxy/model/migrations/alembic/ --ignore=galaxy_test/ --ignore=tool_shed/test/'
     pytest $unit_extra .
```
Comment (nsoranzo) [inline packages/test.sh]:
```suggestion
    # Ignore exit code 5 (no tests ran)
    pytest $unit_extra . || test $? -eq 5
```

File: packages/test.sh (hunk 1 above)
Comment (jmchilton) [reply packages/test.sh]: > If it doesn't have tests, it's probably broken 😄

They're tests 😆... but are the packaged tests broken... yet 😆.

**Comment by jmchilton** [comment]
I would have spent hours on that shell tweak - thanks @nsoranzo !

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**